packages/waku/src/lib/plugins/vite-plugin-fs-router-typegen.ts

```diff
@@ -0,0 +1,195 @@
+import type { Plugin } from 'vite';
+import { readdir, writeFile } from 'node:fs/promises';
+import { existsSync, readFileSync } from 'node:fs';
+import path from 'node:path';
+import { SRC_ENTRIES, SRC_PAGES } from '../constants.js';
+import { joinPath } from '../utils/path.js';
+
+const srcToName = (src: string) => {
+  const split = src
+    .split('/')
+    .map((part) => (part[0]!.toUpperCase() + part.slice(1)).replace('-', '_'));
+
+  if (src.endsWith('_layout.tsx')) {
+    return split.slice(0, -1).join('') + '_Layout';
+  } else if (src.endsWith('index.tsx')) {
+    return split.slice(0, -1).join('') + 'Index';
+  } else if (split.at(-1)?.startsWith('[...')) {
+    const fileName = split
+      .at(-1)!
+      .replace('.tsx', '')
+      .replace('[...', '')
+      .replace(']', '');
+    return (
+      split.slice(0, -1).join('') +
+      'Wild' +
+      fileName[0]!.toUpperCase() +
+      fileName.slice(1)
+    );
+  } else if (split.at(-1)?.startsWith('[')) {
+    const fileName = split
+      .at(-1)!
+      .replace('.tsx', '')
+      .replace('[', '')
+      .replace(']', '');
+    return (
+      split.slice(0, -1).join('') +
+      'Slug' +
+      fileName[0]!.toUpperCase() +
+      fileName.slice(1)
+    );
+  } else {
+    const fileName = split.at(-1)!.replace('.tsx', '');
+    return (
+      split.slice(0, -1).join('') +
+      fileName[0]!.toUpperCase() +
+      fileName.slice(1)
+    );
+  }
+};
+
+export const fsRouterTypegenPlugin = (opts: { srcDir: string }): Plugin => {
+  let entriesFile: string | undefined;
+  let pagesDir: string | undefined;
+  let outputFile: string | undefined;
+  let formatter = (s: string): Promise<string> => Promise.resolve(s);
+  return {
+    name: 'vite-plugin-fs-router-typegen',
+    apply: 'serve',
+    async configResolved(config) {
+      pagesDir = joinPath(config.root, opts.srcDir, SRC_PAGES);
+      entriesFile = joinPath(config.root, opts.srcDir, `${SRC_ENTRIES}.tsx`);
```

Comment:
entriesFile can have different extensions such as `.ts` and `.js`. Please leave a TODO comment unless it's easy to fix.

Reply:
updated to check for ts, tsx, js, and jsx

easy change :)

Reply:
https://github.com/dai-shi/waku/pull/886#discussion_r1778088965